### ISSUE TITLE
npm broken in latest OS X builds

### ISSUE THREAD

**Issue by ebolyen** [conda-forge/nodejs-feedstock#50]
It appears that the npm binary is no longer working on OS X, we noticed this because we use this conda package in a build process for other conda packages while compiling static assets. This used to work, but now we get this error:
```
(npm) ghost:~ caporasolab$ npm
internal/modules/cjs/loader.js:583
    throw err;
    ^

Error: Cannot find module '../lib/utils/unsupported.js'

```

From googling, it looks like this is a relatively common issue on OS X, but it's usually related to homebrew. I've confirmed that node/npm are not installed at all on the host system.

<br/>
Environment (<code>conda list</code>):
<details>

```
(npm) ghost:~ caporasolab$ conda list
# packages in environment at /Users/caporasolab/miniconda3/envs/npm:
#
# Name                    Version                   Build  Channel
nodejs                    10.8.0               hfc679d8_1    conda-forge

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
(npm) ghost:~ caporasolab$ conda info

     active environment : npm
    active env location : /Users/caporasolab/miniconda3/envs/npm
            shell level : 1
       user config file : /Users/caporasolab/.condarc
 populated config files : 
          conda version : 4.5.11
    conda-build version : 3.8.1
         python version : 3.6.5.final.0
       base environment : /Users/caporasolab/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/osx-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /Users/caporasolab/miniconda3/pkgs
                          /Users/caporasolab/.conda/pkgs
       envs directories : /Users/caporasolab/miniconda3/envs
                          /Users/caporasolab/.conda/envs
               platform : osx-64
             user-agent : conda/4.5.11 requests/2.18.4 CPython/3.6.5 Darwin/17.3.0 OSX/10.13.2
                UID:GID : 502:20
             netrc file : None
           offline mode : False


```
</details>

**Comment by minrk** [comment]
Hmm, tricky. I've got:


Environment (<code>conda list</code>):
<details>

```
(test-npm) minrk[17:12] $ conda list
# packages in environment at /Users/minrk/conda/envs/test-npm:
#
# Name                    Version                   Build  Channel
nodejs                    10.8.0               hfc679d8_1    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
(test-npm) minrk[17:12] $ conda info

     active environment : test-npm
    active env location : /Users/minrk/conda/envs/test-npm
            shell level : 1
       user config file : /Users/minrk/.condarc
 populated config files : /Users/minrk/.condarc
          conda version : 4.5.11
    conda-build version : 3.10.9
         python version : 3.6.5.final.0
       base environment : /Users/minrk/conda  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/osx-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /Users/minrk/conda/pkgs
                          /Users/minrk/.conda/pkgs
       envs directories : /Users/minrk/conda/envs
                          /Users/minrk/.conda/envs
               platform : osx-64
             user-agent : conda/4.5.11 requests/2.19.1 CPython/3.6.5 Darwin/18.0.0 OSX/10.14
                UID:GID : 501:20
             netrc file : /Users/minrk/.netrc
           offline mode : False

```
</details>

which is awfully similar, but no errors with npm. Do you perhaps have anything suspicious on any \*PATH env variables? What about `which -a npm` and `which -a node`?

**Comment by ebolyen** [comment]
Hey @minrk,

Thanks for taking a look, since you weren't able to reproduce I looked a little harder and discovered that `conda clean --all` wasn't clearing out the `pkgs` cache like it normally did. After deleting it and re-installing the build, everything worked as expected.

Looks like I had just pulled down a corrupted package from Anaconda and conda continued to re-link the broken package. Sorry, and thanks!
